File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        HashMap<Integer, Integer> hm = new HashMap();
        for(int i=0; i<nums.length; i++) {
            int temp = target - nums[i];
            if(hm.containsKey(temp)) {
                ans[0] = i;
                ans[1] = hm.get(temp);
                return ans;
            }
            hm.put(nums[i], i);
        }
        
        return ans;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        vector<int> ans;
        for(int i=0; i<nums.size(); i++) {
            int t = target - nums[i];
            if(map.find(t) != map.end()) {
                ans.push_back(i);
                ans.push_back(map[t]);
                break;
            }
            map[nums[i]] = i;
        }
        return ans;
    }
};